### PR TITLE
Clarify ssl usage for Sensu Go

### LINKED ISSUE #1251: Configuration parameters for certificates should describe what the certificate is used for
## This is an issue with:

- [ ] Bug (site functionality or styling)
- [ ] Errata (fix needed for doc content)
- [ ] New content (guide wanted)
- [x] Update (Add missing or refresh existing content)
- [ ] Enhancement (add new site functionality)

## Expected Behavior

Reference documentation for each parameter related to an encryption certificate describes what the certificate is used for.

## Current Behavior

Parameters related to encryption certificates in Backend and Agent references do not describe the specific use of each certificate.

## Context

We believe this will help to satisfy questions arising from folks reading through the Securing Sensu guide as well as external security reviews.

### PR DESCRIPTION
**Pull request by asachs01**
## Description

Closes #1251 by elaborating on SSL usage.

### CODE REVIEW

File: content/sensu-go/5.0/reference/backend.md [hunk 1]
```diff
@@ -428,7 +428,7 @@ etcd-advertise-client-urls:
 
 | etcd-cert-file |      |
 -----------------|------
-description      | Path to the client server TLS cert file
+description      | Path to the client server TLS cert file. Secures communication with the etcd client server.
```
Comment (cwjohnston) [inline content/sensu-go/5.0/reference/backend.md]: In this case what is "the etcd client server"?

I think this and many similar descriptions in the PR can be made more clear by specifying which things the communication is between.

```suggestion
description      | Path to the etcd client API TLS cert file. Secures communication between the embedded etcd client API and any etcd clients.
```

File: content/sensu-go/5.0/reference/backend.md [hunk 2]
```diff
@@ -338,10 +338,10 @@ agent-port: 8081{{< /highlight >}}
 
 | cert-file  |      |
 -------------|------
-description  | SSL/TLS certificate
+description  | SSL/TLS certificate. This certificate secures communication with the Sensu Dashboard and API.
```
Comment (cwjohnston) [inline content/sensu-go/5.0/reference/backend.md]:
```suggestion
description  | Path to the primary backend certificate file. This certificate secures communications between Sensu Dashboard and end user web browsers, as well as communication between sensuctl and the Sensu API.
```

File: content/sensu-go/5.0/reference/backend.md [hunk 3]
```diff
@@ -504,7 +504,7 @@ etcd-initial-cluster-token: "sensu"{{< /highlight >}}
 
 | etcd-key-file  |      |
 -----------------|------
-description      | Path to the client server TLS key file
+description      | Path to the client server TLS key file. Secures communication with the etcd client server.
```
Comment (cwjohnston) [inline content/sensu-go/5.0/reference/backend.md]:
```suggestion
description      | Path to the etcd client API TLS key file. Secures communication between the embedded etcd client API and any etcd clients.
```

File: content/sensu-go/5.0/reference/backend.md [hunk 4]
```diff
@@ -582,7 +582,7 @@ etcd-peer-client-cert-auth: true{{< /highlight >}}
 
 | etcd-peer-key-file |      |
 ---------------------|------
-description          | Path to the peer server TLS key file
+description          | Path to the peer server TLS key file. Secures communication with the etcd peer server.
```
Comment (cwjohnston) [inline content/sensu-go/5.0/reference/backend.md]:
```suggestion
description          | Path to the etcd peer API TLS key file. Secures communication between etcd cluster members.
```

File: content/sensu-go/5.0/reference/backend.md [hunk 5]
```diff
@@ -593,7 +593,7 @@ etcd-peer-key-file: "./backend-0-key.pem"{{< /highlight >}}
 
 | etcd-peer-trusted-ca-file |      |
```
Comment (cwjohnston) [inline content/sensu-go/5.0/reference/backend.md:594]:
```suggestion
| etcd-peer-cert-file |      |
```

File: content/sensu-go/5.0/reference/backend.md [hunk 6]
```diff
@@ -593,7 +593,7 @@ etcd-peer-key-file: "./backend-0-key.pem"{{< /highlight >}}
...
 ----------------------------|------
-description                 | Path to the peer server TLS key file
+description                 | Path to the peer server TLS key file. Secures communication with the etcd peer server.
```
Comment (cwjohnston) [inline content/sensu-go/5.0/reference/backend.md]:
```suggestion
description                 | Path to the peer server TLS certificate file. This certificate secures communication between etcd cluster members.
```

**Review by cwjohnston — DISMISSED** [review]

File: content/sensu-go/5.0/reference/backend.md [hunk 7]
```diff
@@ -338,10 +338,10 @@ agent-port: 8081{{< /highlight >}}
 
 | cert-file  |      |
 -------------|------
-description  | SSL/TLS certificate
+description  | Path to the primary backend certificate file. This certificate secures communications between Sensu Dashboard and end user web browsers, as well as communication between sensuctl and the Sensu API.
 type         | String
 default      | `""`
-example      | {{< highlight shell >}}# Command line example
+  example      | {{< highlight shell >}}# Command line example
```
Comment (apaskulin) [inline content/sensu-go/5.0/reference/backend.md]:
```suggestion
example      | {{< highlight shell >}}# Command line example
```

File: content/sensu-go/5.1/reference/backend.md [hunk 8]
```diff
@@ -560,7 +560,7 @@ etcd-name: "backend-0"{{< /highlight >}}
 
 | etcd-peer-cert-file |      |
 ----------------------|------
-description           | Path to the peer server TLS cert file
+description           |  Path to the peer server TLS certificate file. This certificate secures communication between etcd cluster members.
```
Comment (apaskulin) [inline content/sensu-go/5.1/reference/backend.md]:
```suggestion
description           | Path to the peer server TLS certificate file. This certificate secures communication between etcd cluster members.
```

**Review by apaskulin — COMMENTED** [review]
Two suggestions, then :shipit:

**Review by apaskulin — APPROVED** [review]
🙌

**Pull request merged**